Lib/asyncio/locks.py

```diff
@@ -3,96 +3,13 @@
 __all__ = ('Lock', 'Event', 'Condition', 'Semaphore', 'BoundedSemaphore')
 
 import collections
-import types
 import warnings
 
 from . import events
-from . import futures
 from . import exceptions
-from .import coroutines
-
-
-class _ContextManager:
-    """Context manager.
-
-    This enables the following idiom for acquiring and releasing a
-    lock around a block:
-
-        with (yield from lock):
-            <block>
-
-    while failing loudly when accidentally using:
-
-        with lock:
-            <block>
-
-    Deprecated, use 'async with' statement:
-        async with lock:
-            <block>
-    """
-
-    def __init__(self, lock):
-        self._lock = lock
-
-    def __enter__(self):
-        # We have no use for the "as ..."  clause in the with
-        # statement for locks.
-        return None
-
-    def __exit__(self, *args):
-        try:
-            self._lock.release()
-        finally:
-            self._lock = None  # Crudely prevent reuse.
 
 
 class _ContextManagerMixin:
-    def __enter__(self):
-        raise RuntimeError(
-            '"yield from" should be used as context manager expression')
```

Comment:
Should we keep this? We can change the error message to say that `async with should be used...`

Reply:
I think the motivation was emphasizing that context managers were supported but in a slightly unusual way: `with (yield from lock):`; bare `with lock:` was forbidden.

After switching to `async with lock:` everything becomes clear: `with` is not supported in any form for the sake of `async with`. The error message is standard and well-known: `AttributeError: __enter__`.  I think we shouldn't replace the default behavior.

Maybe the standard text can be improved globally, but it is another story, I guess.